classes/Hook.php

```diff
@@ -1244,4 +1250,30 @@ private static function getMethodName(string $hookName): string
     {
         return 'hook' . ucfirst($hookName);
     }
+
+    /**
+     * Return an instance of the specified hook.
+     *
+     * @param string $hook_name Hook name
+     *
+     * @return bool
+     */
+    public static function getStatusByName($hook_name): bool
+    {
+        if (empty($hook_name)) {
+            return true;
```

Comment:
Should be false, the hook is empty, it should not be considered as active :thinking:

Reply:
To avoid unexpected behavior ... Return false in this case can introduce weird behavior, For example this exception will no longer be triggered :   https://github.com/PrestaShop/PrestaShop/blob/develop/classes/Hook.php#L792

Reply:
An exception can be raised here otherwise. The function must be consistent and returns the correct value when the hook is empty or no hook is found. Otherwise, in the future, we will don't understand why this function return true whereas nothing is good in the name :sweat_smile:

Reply:
I understand you perfectly! I'll find another way to do it avoiding BC, if you have any suggestions I'm available to discuss them!